File: Nodejs/K10.ts
import { IExecutable } from "./IExecutable"
import * as readline from "readline"

export class K10 extends IExecutable {
    Question1(rl?: readline.ReadLine): Promise<string> {
        let inputString: string;

        let q1 = () => {
            return new Promise((resolve, reject) => {
                rl.question("文字列の入力＞", (value) => {
                    inputString = value;
                    resolve();
                });
            });
        };

        return new Promise(async (resolve, reject) => {
            await q1();
            console.log(`入力された文字列の文字数：${inputString.length}`);
            resolve();
        });
    }

    Question2(rl?: readline.ReadLine): Promise<string> {
        let str = "STRINGcopy";
        console.log(`コピーしたい文字列：${str}`);
        let copyStr = str;
        console.log(`コピー後の文字列：${copyStr}`);

        return new Promise((resolve, reject) => {
            resolve();
        });
    }

    Question3(rl?: readline.ReadLine): Promise<string> {
        let str = [
            "abcdef", "GHIJKL", "13579"
        ];

        let strCombined = "";

        console.log("連結したい３つの文字列：");
        for (var s of str) {
            console.log(s);
            strCombined += s;
        }

        console.log(`連結した文字列：\n${strCombined}`);

        return new Promise((resolve, reject) => {
            resolve();
        });
    }

    Question4(rl?: readline.ReadLine): Promise<string> {
        let alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        let inputString = "I AM A STUDENT. I AM GOOD AT GAME PROGRAMMING.";

        let count = 0;
        let _count = 0;
        console.log(`文字列：${inputString}\n`);
        console.log("文字頻度");
        for (var i = 0; i < alphabets.length; i++) {
            for (var j = 0; j < inputString.length; j++) {
                if (alphabets[i] === inputString[j]) {
                    count++;
                }
            }
            process.stdout.write(`${alphabets[i]}: ${count}\t`);
            count = 0;
            _count++;

            if (_count % 4 === 0) {
                console.log();
            }
        }

        return new Promise((resolve, reject) => {
            resolve();
        });
    }


}

File: Cpp/K10.cpp
#include <array>
#include <iostream>
#include "K10.h"

using namespace std;

K10::K10() = default;


K10::~K10() = default;

void K10::question1() {
    string inputString;
    cout << "文字列の入力＞";
    cin >> inputString;

    cout << "入力された文字列の文字数：" << inputString.length();
}

void K10::question2() {
    string str = "STRINGcopy";

    cout << "コピーしたい文字列：" << str << "\n";

    string copyStr{str};

    cout << "コピー後の文字列：" << copyStr << "\n";

}

void K10::question3() {

    array<string, 3> str = {
            "abcdef", "GHIJKL", "13579"
    };

    string str_combined;

    cout << "連結したい３つの文字列：\n";
    for (const auto &_str: str) {
        str_combined += _str;
    }

    cout << "連結した文字列：\n" << str_combined;
}

void K10::question4() {
    string alphabets = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    string inputString = "I AM A STUDENT. I AM GOOD AT GAME PROGRAMMING.";

    cout << "文字列：" << inputString << "\n\n";

    cout << "文字頻度\n";

    auto count = 0;
    auto _count = 0;

    auto i = 0;

    for (char alphabet: alphabets) {
        for (char j: inputString) {
            if (alphabet == j) {
                ++_count;
            }
        }

        cout << alphabet << ": " << _count << "\t";
        _count = 0;
        ++count;

        if (count % 4 == 0) {
            printf("\n");
        }
    }
}
